File: app/src/main/java/com/hippo/ehviewer/dao/DownloadLabel.java
package com.hippo.ehviewer.dao;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "DOWNLOAD_LABELS")
public class DownloadLabel {

    @PrimaryKey
    @ColumnInfo(name = "_id")
    private Long id;
    @ColumnInfo(name = "LABEL")
    private String label;
    @ColumnInfo(name = "TIME")
    private long time;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

}

File: app/src/main/java/com/hippo/ehviewer/dao/DownloadLabel.kt
package com.hippo.ehviewer.dao

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "DOWNLOAD_LABELS", indices = [Index("LABEL", unique = true)])
data class DownloadLabel(
    @ColumnInfo(name = "LABEL")
    var label: String,

    @ColumnInfo(name = "POSITION")
    var position: Int,

    @PrimaryKey
    @ColumnInfo(name = "_id")
    var id: Long? = null,
)
